sklearn/utils/fixes.py

```diff
@@ -107,22 +108,25 @@ def _eigh(*args, **kwargs):
 
 
 # remove when https://github.com/joblib/joblib/issues/1071 is fixed
-def delayed(function):
+def delayed(func, thread=threading.current_thread()):
     """Decorator used to capture the arguments of a function."""
 
-    @functools.wraps(function)
-    def delayed_function(*args, **kwargs):
-        return _FuncWrapper(function), args, kwargs
+    def decorate(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            return _FuncWrapper(func, thread=thread), args, kwargs
 
-    return delayed_function
+        return wrapper
+
+    return decorate(func)
 
 
 class _FuncWrapper:
     """ "Load the global configuration before calling the function."""
 
-    def __init__(self, function):
+    def __init__(self, function, thread):
         self.function = function
-        self.config = get_config()
+        self.config = get_config(thread=thread)
```

Comment:
I'm concerned about adding more API to `get_config`. If we want to avoid more API, then we can call `_get_thread_config(thread=thread)` here instead.